SQF/dayz_code/compile/doorManagement/fn_check_access.sqf

```diff
@@ -0,0 +1,146 @@
+	
+	Parameters: 
+	_this select 0: object - player calling this function
+	_this select 1: object - target to check ownership and friendlies of
+	
+	Returns:
+	_return select 0: bool - player is owner of target object
+	_return select 1: bool - player is friends with owner of target object
+	_return select 2: bool - player is plot owner
+	_return select 3: bool - player is plot friend
+	_return select 4: bool - player is plot management admin
+	_return select 5: bool - player is target friend
+	_return select 6: bool - player is target management admin
+	_return select 7: string - detected target type
+	
+
+	2,3,4: always false if no plot nearby
+	5,6: always false if _target is not a supported target
+*/
+
+private [
+	 "_player"						// Player, who wants to access the _target
+	,"_target"						// The target object the _player wants to access
+	,"_isOwner"						// return value
+	,"_isFriendly"					// return value
+	,"_isPlotOwner"					// return value
+	,"_isPlotFriend"				// return value
+	,"_isPlotAdmin"					// return value
+	,"_isTargetFriend"				// return value
+	,"_isTargetAdmin"				// return value
+	,"_targetType"					// return value
+	,"_targetOwnerUID"				// UID of the owner of _target
+	,"_playerUID"					// UID of the _player
+	,"_plotcheck"					// takes return value of FNC_find_plots
+	,"_isNearPlot"					// player is in plot's reach
+	,"_nearestPlot"					// plot object
+	,"_plotOwnerUID"				// plot owner's UID
+	,"_plotFriends"					// list of plot friends [["UID", "Name"], ..]
+	,"_targetFriends"				// list of target friends [["UID", "Name"], ..]
+];
+
+
+// assign parameters
+_player = _this select 0;
+_target = _this select 1;
+
+// Initialize return values
+_isOwner = false;
+_isFriendly = false;
+_isPlotOwner = false;
+_isPlotFriend = false;
+_isPlotAdmin = false;
+_isTargetFriend = false;
+_isTargetAdmin = false;
+_targetType = "";
+
+
+// determine target type
+_targetType = if(typeOf _target in DZE_DoorsLocked) then { "DOOR"; } else { "GENERIC"; };
+
+
+// determine owner and player id
+// and check if player is owner of target object
+if(DZE_permanentPlot) then {
+	_targetOwnerUID = _target getVariable ["ownerPUID","0"];
+	_playerUID = [_player] call FNC_GetPlayerUID;
+} else {
+	_targetOwnerUID = _target getVariable ["characterID","0"];
+	_playerUID = dayz_characterID;
+};
+_isOwner = (_playerUID == _targetOwnerUID);
+
+
+// determine _players friends (tagged)
+// and check if owner of _target is tagged friendly
+_friendlies	= _player getVariable ["friendlyTo",[]];
+_isFriendly = (_targetOwnerUID in _friendlies);
+
+
+// find nearest plot
+_plotcheck = [_player, false] call FNC_find_plots;
+_isNearPlot = ((_plotcheck select 1) > 0);
+_nearestPlot = _plotcheck select 2;
+if(_isNearPlot) then {
+	// determine plot owner
+	// and check if player is owner of plot
+	if (DZE_permanentPlot) then {
+		_plotOwnerUID = _nearestPlot getVariable ["ownerPUID","0"];
+	} else {
+		_plotOwnerUID = _target getVariable ["characterID","0"];
```

Comment:
_target should be _nearestPlot here. The CharacterID of the door may not be the same as the CharacterID of the nearest plot pole.